src/main/java/core/basesyntax/service/operation/ReturnOperation.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.dao.FruitTransactionDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class ReturnOperation implements OperationHandler {
+    private final FruitTransactionDao fruitTransactionDao;
+
+    public ReturnOperation(FruitTransactionDao fruitTransactionDao) {
+        this.fruitTransactionDao = fruitTransactionDao;
+    }
+    @Override
+    public void recount(FruitTransaction transaction) {
+        if (transaction.getQuantity() < 0) {
+            throw new RuntimeException("Quantity can't be less than 0.");
+        }
+        int currentQuantity = fruitTransactionDao.get(transaction.getFruit());
+        currentQuantity += transaction.getQuantity();
+        fruitTransactionDao.update(transaction.getFruit(), currentQuantity);
```

Comment:
Direct manipulation of the storage inside the operation handler violates the encapsulation principle. Consider creating a service that abstracts the access and modification of the storage, keeping the operation handler focused on the business logic and not on the data access layer.